client/lib/push-notifications/service-worker.js

```diff
@@ -0,0 +1,20 @@
+/**
+ *  This file is served as-is as /service-worker.js
+ **/
+
+/* eslint-disable */
+'use strict';
+/* eslint-enable */
+
+self.addEventListener( 'push', function( event ) {
+	var notification = event.data.json();
```

Comment:
It's usually a good idea to do a check input before executing to prevent fatal errors. Something like:
```
if ( typeof event.data !== 'object' && typeof event.data.json !== 'function' ) {
    return;
}
```